src/main/java/org/truffleruby/core/time/RubyDateFormatter.java

```diff
@@ -544,6 +545,130 @@ public static RopeBuilder formatToRopeBuilder(List<Token> compiledPattern, Zoned
         return toAppendTo;
     }
 
+    public static boolean formatToRopeBuilderCanBeFast(Token[] compiledPattern) {
+        for (Token token : compiledPattern) {
+            Format format = token.getFormat();
+
+            switch (format) {
+                case FORMAT_ENCODING:
+                    // We only care about UTF8 encoding
+                    if (!((Encoding) token.getData() == UTF8Encoding.INSTANCE)) {
+                        return false;
+                    } else {
+                        continue;
+                    }
+                case FORMAT_OUTPUT:
```

Comment:
At first, we thought the formatter would be the `DEFAULT_FORMATTER`, but it isn't. The formatter is at 6 width when this part hits.

Reply:
Do we not need to check that then?

Reply:
If we have the check, then our case would return false (because it doesn't have the default formatter) and we won't use the fast path.

Reply:
Right - I mean if it's not the default we should be checking for - let's check it's whatever it actually is and then hard-code for that.